app/core/ts/utils/io/Importer.ts

```diff
@@ -2,11 +2,14 @@ import {SAVED} from "../Config";
 
 import {XMLReader} from "./xml/XMLReader";
 import {ResolveVersionConflict} from "./VersionConflictResolver";
-import {DigitalCircuitDesigner} from "digital/models/DigitalCircuitDesigner";
+import {DigitalCircuitController} from "site/digital/controllers/DigitalCircuitController";
+import {Vector} from "Vector";
 
 export const Importer = (() => {
     return {
-        LoadCircuit: function(designer: DigitalCircuitDesigner, fileContents: string | XMLDocument): string {
+        LoadCircuit: function(main: DigitalCircuitController, fileContents: string | XMLDocument,loadCamera: boolean): string {
```

Comment:
see above comment about taking in camera

should be
`LoadCircuit: function(designer: DigitalCircuitDesigner, fileContents: string | XMLDocument, camera?: Camera)`